src/renderers/shaders/ShaderChunk/shadowmap_fragment.glsl

```diff
@@ -26,148 +41,252 @@
 
 		bool frustumTest = all( frustumTestVec );
 
-		if ( frustumTest ) {
-
-			shadowCoord.z += shadowBias[ i ];
+		if ( frustumTest || isPointLight ) {			
 
 			#if defined( SHADOWMAP_TYPE_PCF )
 
+				#if defined(POINT_LIGHT_SHADOWS)
+
+					if( isPointLight ) {
+
+						float cubeTexelSize = 1.0 / ( shadowMapSize[ i ].x * 0.25 );
+						vec3 baseDirection3D = normalize( lightToPosition );
+						vec2 baseDirection2D = cubeToUV( baseDirection3D, texelSizeX, texelSizeY );
+
+						initGridSamplingDisk();
+
+						float diskRadius = 1.25;
+						float numSamples = 1.0;
+						shadow = 0.0;
+
+						vec3 baseDirection = normalize( lightToPosition );
+						float curDistance = length( lightToPosition );
+
+						float dist = unpack1K( texture2D( shadowMap[ i ],  baseDirection2D ) ) + 0.1;
+						if ( curDistance >= dist )
+							shadow += 1.0;
+						
+						// evaluate each sampling direction
+						for( int s = 0; s < 20; s++ ) {
+						 
+							vec3 offset = gridSamplingDisk[ s ] * diskRadius * cubeTexelSize;
+							vec3 adjustedBaseDirection3D = baseDirection3D + offset;
+							vec2 adjustedBaseDirection2D = cubeToUV( adjustedBaseDirection3D, texelSizeX, texelSizeY );
+							dist = unpack1K( texture2D( shadowMap[ i ],  adjustedBaseDirection2D ) ) + shadowBias[ i ];
+							if ( curDistance >= dist )
+								shadow += 1.0;
+							numSamples += 1.0;
+
+						}
+
+						shadow /= numSamples;
+
+					} else {
+
+				#endif
+
 						// Percentage-close filtering
 						// (9 pixel kernel)
 						// http://fabiensanglard.net/shadowmappingPCF/
+						
+						/*
+								// nested loops breaks shader compiler / validator on some ATI cards when using OpenGL
+								// must enroll loop manually
+							for ( float y = -1.25; y <= 1.25; y += 1.25 )
+								for ( float x = -1.25; x <= 1.25; x += 1.25 ) {
```

Comment:
Anyone remember details / when it was / has an affected GPU for testing? /ping @WestLangley @mrdoob 
I guess the issue in this comment *may* be specific to the index being of type `float`?